Ingested Data

Statistics Summary
{
  "lastmonth_activity": {
    "mean": 165.65384615384616,
    "median": 73.0,
    "std": 284.0332293669447
  },
  "lastyear_activity": {
    "mean": 1502.923076923077,
    "median": 955.0,
    "std": 2192.6449584568304
  },
  "number_of_employees": {
    "mean": 26.884615384615383,
    "median": 14.0,
    "std": 31.353885785435814
  }
}

Diagnostics Summary
{
  "execution_time": [
    {
      "ingest_time_mean": 0.49363791280009084
    },
    {
      "train_time_mean": 0.9271940053500429
    }
  ],
  "missing_percentage": {
    "corporation": {
      "percentage": 0.0
    },
    "exited": {
      "percentage": 0.0
    },
    "lastmonth_activity": {
      "percentage": 0.0
    },
    "lastyear_activity": {
      "percentage": 0.0
    },
    "number_of_employees": {
      "percentage": 0.0
    }
  },
  "outdated_packages": []
}


Test Data

Model Predictions
[
  0,
  1,
  1,
  1,
  1
]

Model Score
<!doctype html>
<html lang=en>
  <head>
    <title>FileNotFoundError: [Errno 2] No such file or directory: &#39;python&#39;
 // Werkzeug Debugger</title>
    <link rel="stylesheet" href="?__debugger__=yes&amp;cmd=resource&amp;f=style.css">
    <link rel="shortcut icon"
        href="?__debugger__=yes&amp;cmd=resource&amp;f=console.png">
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=debugger.js"></script>
    <script>
      var CONSOLE_MODE = false,
          EVALEX = true,
          EVALEX_TRUSTED = false,
          SECRET = "nSTSsXk09ZQemNHXtXWC";
    </script>
  </head>
  <body style="background-color: #fff">
    <div class="debugger">
<h1>FileNotFoundError</h1>
<div class="detail">
  <p class="errormsg">FileNotFoundError: [Errno 2] No such file or directory: &#39;python&#39;
</p>
</div>
<h2 class="traceback">Traceback <em>(most recent call last)</em></h2>
<div class="traceback">
  <h3></h3>
  <ul><li><div class="frame" id="frame-140303314934064">
  <h4>File <cite class="filename">"/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py"</cite>,
      line <em class="line">1498</em>,
      in <code class="function">__call__</code></h4>
  <div class="source "><pre class="line before"><span class="ws">    </span>) -&gt; cabc.Iterable[bytes]:</pre>
<pre class="line before"><span class="ws">        </span>&#34;&#34;&#34;The WSGI server calls the Flask application object as the</pre>
<pre class="line before"><span class="ws">        </span>WSGI application. This calls :meth:`wsgi_app`, which can be</pre>
<pre class="line before"><span class="ws">        </span>wrapped to apply middleware.</pre>
<pre class="line before"><span class="ws">        </span>&#34;&#34;&#34;</pre>
<pre class="line current"><span class="ws">        </span>return self.wsgi_app(environ, start_response)</pre></div>
</div>

<li><div class="frame" id="frame-140303307683008">
  <h4>File <cite class="filename">"/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py"</cite>,
      line <em class="line">1476</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source "><pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line before"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line before"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line before"><span class="ws">                </span>error = e</pre>
<pre class="line current"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
<pre class="line after"><span class="ws">                </span>raise</pre>
<pre class="line after"><span class="ws">            </span>return response(environ, start_response)</pre>
<pre class="line after"><span class="ws">        </span>finally:</pre></div>
</div>

<li><div class="frame" id="frame-140303307683120">
  <h4>File <cite class="filename">"/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py"</cite>,
      line <em class="line">1473</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source "><pre class="line before"><span class="ws">        </span>ctx = self.request_context(environ)</pre>
<pre class="line before"><span class="ws">        </span>error: BaseException | None = None</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line current"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line after"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">                </span>error = e</pre>
<pre class="line after"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre></div>
</div>

<li><div class="frame" id="frame-140303307683232">
  <h4>File <cite class="filename">"/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py"</cite>,
      line <em class="line">882</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source "><pre class="line before"><span class="ws">            </span>request_started.send(self, _async_wrapper=self.ensure_sync)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line before"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line before"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line current"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(</pre>
<pre class="line after"><span class="ws">        </span>self,</pre>
<pre class="line after"><span class="ws">        </span>rv: ft.ResponseReturnValue | HTTPException,</pre></div>
</div>

<li><div class="frame" id="frame-140303307683344">
  <h4>File <cite class="filename">"/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py"</cite>,
      line <em class="line">880</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source "><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>request_started.send(self, _async_wrapper=self.ensure_sync)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line current"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line after"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(</pre></div>
</div>

<li><div class="frame" id="frame-140303307683456">
  <h4>File <cite class="filename">"/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py"</cite>,
      line <em class="line">865</em>,
      in <code class="function">dispatch_request</code></h4>
  <div class="source "><pre class="line before"><span class="ws">            </span>and req.method == &#34;OPTIONS&#34;</pre>
<pre class="line before"><span class="ws">        </span>):</pre>
<pre class="line before"><span class="ws">            </span>return self.make_default_options_response()</pre>
<pre class="line before"><span class="ws">        </span># otherwise dispatch to the handler for that endpoint</pre>
<pre class="line before"><span class="ws">        </span>view_args: dict[str, t.Any] = req.view_args  # type: ignore[assignment]</pre>
<pre class="line current"><span class="ws">        </span>return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def full_dispatch_request(self) -&gt; Response:</pre>
<pre class="line after"><span class="ws">        </span>&#34;&#34;&#34;Dispatches the request and on top of that performs request</pre>
<pre class="line after"><span class="ws">        </span>pre and postprocessing as well as HTTP exception catching and</pre>
<pre class="line after"><span class="ws">        </span>error handling.</pre></div>
</div>

<li><div class="frame" id="frame-140303307683568">
  <h4>File <cite class="filename">"/mnt/c/Users/rdutr/Documents/Projects/Udacity/Dynamic-Risk-Assessment-System/src/app.py"</cite>,
      line <em class="line">48</em>,
      in <code class="function">score</code></h4>
  <div class="source "><pre class="line before"><span class="ws">    </span>gets the score of the deployed model</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span>Returns:</pre>
<pre class="line before"><span class="ws">        </span>str: model f1 score</pre>
<pre class="line before"><span class="ws">    </span>&#34;&#34;&#34;</pre>
<pre class="line current"><span class="ws">    </span>output = subprocess.run([&#39;python&#39;, &#39;scoring.py&#39;],</pre>
<pre class="line after"><span class="ws">                            </span>capture_output=True).stdout</pre>
<pre class="line after"><span class="ws">    </span>output = re.findall(r&#39;f1 score = \d*\.?\d+&#39;, output.decode())[0]</pre>
<pre class="line after"><span class="ws">    </span>return output</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span> </pre></div>
</div>

<li><div class="frame" id="frame-140303307685360">
  <h4>File <cite class="filename">"/usr/lib/python3.10/subprocess.py"</cite>,
      line <em class="line">503</em>,
      in <code class="function">run</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>raise ValueError(&#39;stdout and stderr arguments may not be used &#39;</pre>
<pre class="line before"><span class="ws">                             </span>&#39;with capture_output.&#39;)</pre>
<pre class="line before"><span class="ws">        </span>kwargs[&#39;stdout&#39;] = PIPE</pre>
<pre class="line before"><span class="ws">        </span>kwargs[&#39;stderr&#39;] = PIPE</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line current"><span class="ws">    </span>with Popen(*popenargs, **kwargs) as process:</pre>
<pre class="line after"><span class="ws">        </span>try:</pre>
<pre class="line after"><span class="ws">            </span>stdout, stderr = process.communicate(input, timeout=timeout)</pre>
<pre class="line after"><span class="ws">        </span>except TimeoutExpired as exc:</pre>
<pre class="line after"><span class="ws">            </span>process.kill()</pre>
<pre class="line after"><span class="ws">            </span>if _mswindows:</pre></div>
</div>

<li><div class="frame" id="frame-140303307809488">
  <h4>File <cite class="filename">"/usr/lib/python3.10/subprocess.py"</cite>,
      line <em class="line">971</em>,
      in <code class="function">__init__</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">                </span>self.stderr = io.open(errread, &#39;rb&#39;, bufsize)</pre>
<pre class="line before"><span class="ws">                </span>if self.text_mode:</pre>
<pre class="line before"><span class="ws">                    </span>self.stderr = io.TextIOWrapper(self.stderr,</pre>
<pre class="line before"><span class="ws">                            </span>encoding=encoding, errors=errors)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line current"><span class="ws">            </span>self._execute_child(args, executable, preexec_fn, close_fds,</pre>
<pre class="line after"><span class="ws">                                </span>pass_fds, cwd, env,</pre>
<pre class="line after"><span class="ws">                                </span>startupinfo, creationflags, shell,</pre>
<pre class="line after"><span class="ws">                                </span>p2cread, p2cwrite,</pre>
<pre class="line after"><span class="ws">                                </span>c2pread, c2pwrite,</pre>
<pre class="line after"><span class="ws">                                </span>errread, errwrite,</pre></div>
</div>

<li><div class="frame" id="frame-140303307809600">
  <h4>File <cite class="filename">"/usr/lib/python3.10/subprocess.py"</cite>,
      line <em class="line">1863</em>,
      in <code class="function">_execute_child</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">                        </span>err_filename = cwd</pre>
<pre class="line before"><span class="ws">                    </span>else:</pre>
<pre class="line before"><span class="ws">                        </span>err_filename = orig_executable</pre>
<pre class="line before"><span class="ws">                    </span>if errno_num != 0:</pre>
<pre class="line before"><span class="ws">                        </span>err_msg = os.strerror(errno_num)</pre>
<pre class="line current"><span class="ws">                    </span>raise child_exception_type(errno_num, err_msg, err_filename)</pre>
<pre class="line after"><span class="ws">                </span>raise child_exception_type(err_msg)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>def _handle_exitstatus(self, sts,</pre>
<pre class="line after"><span class="ws">                               </span>waitstatus_to_exitcode=os.waitstatus_to_exitcode,</pre></div>
</div>
</ul>
  <blockquote>FileNotFoundError: [Errno 2] No such file or directory: &#39;python&#39;
</blockquote>
</div>

<div class="plain">
    <p>
      This is the Copy/Paste friendly version of the traceback.
    </p>
    <textarea cols="50" rows="10" name="code" readonly>Traceback (most recent call last):
  File &#34;/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py&#34;, line 1498, in __call__
    return self.wsgi_app(environ, start_response)
  File &#34;/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py&#34;, line 1476, in wsgi_app
    response = self.handle_exception(e)
  File &#34;/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py&#34;, line 1473, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py&#34;, line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py&#34;, line 880, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py&#34;, line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File &#34;/mnt/c/Users/rdutr/Documents/Projects/Udacity/Dynamic-Risk-Assessment-System/src/app.py&#34;, line 48, in score
    output = subprocess.run([&#39;python&#39;, &#39;scoring.py&#39;],
  File &#34;/usr/lib/python3.10/subprocess.py&#34;, line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File &#34;/usr/lib/python3.10/subprocess.py&#34;, line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File &#34;/usr/lib/python3.10/subprocess.py&#34;, line 1863, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: &#39;python&#39;
</textarea>
</div>
<div class="explanation">
  The debugger caught an exception in your WSGI application.  You can now
  look at the traceback which led to the error.  <span class="nojavascript">
  If you enable JavaScript you can also use additional features such as code
  execution (if the evalex feature is enabled), automatic pasting of the
  exceptions and much more.</span>
</div>
      <div class="footer">
        Brought to you by <strong class="arthur">DON'T PANIC</strong>, your
        friendly Werkzeug powered traceback interpreter.
      </div>
    </div>

    <div class="pin-prompt">
      <div class="inner">
        <h3>Console Locked</h3>
        <p>
          The console is locked and needs to be unlocked by entering the PIN.
          You can find the PIN printed out on the standard output of your
          shell that runs the server.
        <form>
          <p>PIN:
            <input type=text name=pin size=14>
            <input type=submit name=btn value="Confirm Pin">
        </form>
      </div>
    </div>
  </body>
</html>

<!--

Traceback (most recent call last):
  File "/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py", line 1498, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py", line 1476, in wsgi_app
    response = self.handle_exception(e)
  File "/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/rdutra/.local/lib/python3.10/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/mnt/c/Users/rdutr/Documents/Projects/Udacity/Dynamic-Risk-Assessment-System/src/app.py", line 48, in score
    output = subprocess.run(['python', 'scoring.py'],
  File "/usr/lib/python3.10/subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1863, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'python'


-->
